### PR TITLE
Add simple filters for collections

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

Depends on #5557 

Added simple filter queries for collections in the server API. Such queries should cover simpler cases, while more complex filters should use the JSON syntax.

- Added simple filters for collection endpoints in API (e.g. `/tasks?project_id=42&user=myusername`)
- Removed duplicating collection endpoints in API in favor of their full versions: 
  - `/projects/{id}/tasks` -> `/tasks?project_id={id}`
  - `/tasks/{id}/jobs` -> `/jobs?task_id={id}`
  - `/jobs/{id}/issues` -> `/issues?job_id={id}`
  - `/issue/{id}/comments` -> `/comments?issue_id={id}`
  - Corresponding owning objects now return a link to the collection:
    - `/projects/{id}`: `.tasks`
    - `/tasks/{id}`: `.jobs`
    - `/jobs/{id}`: `.issues`
    - `/issue/{id}`: `.comments`
- Fixed errors in the generated server API schema:
  - Input and output types in all the basic model methods (`Create`, `Update`, `Retrieve`, `Delete`)
  - Removed invalid separation for the project list operation response type
- File structure for projects in the dataset cache is updated to store the new project representation. Old caches will be invalidated.
- Added tests

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # Copyright (C) 2021-2022 Intel Corporation
-# Copyright (C) 2022 CVAT.ai Corporation
```
Comment (nmanovic) [inline cvat/apps/dataset_manager/tests/test_rest_api_formats.py:2]: Let's keep the header

File: cvat/apps/webhooks/serializers.py [hunk 2]
```diff
@@ -93,6 +93,10 @@ class Meta:
             "last_delivery_date",
         )
         read_only_fields = fields
+        extra_kwargs = {
+            "project": {"allow_null": True},
```
Comment (nmanovic) [inline cvat/apps/webhooks/serializers.py:97]: Why do we need "project"? I see `project_id` in the list above of available fields.

File: cvat/apps/webhooks/serializers.py [hunk 3]
```diff
@@ -145,3 +149,6 @@ class Meta:
             "response",
         )
         read_only_fields = fields
+        extra_kwargs = {
```
Comment (nmanovic) [inline cvat/apps/webhooks/serializers.py:152]: I see `status_code = models.PositiveIntegerField(null=True, default=None)` definition in the models.py file. Why do we need it here as well?

Comment (zhiltsov-max) [reply cvat/apps/webhooks/serializers.py]: Hmm, seems like a mistake. Fixed.

File: cvat/apps/webhooks/serializers.py (hunk 2 above)
Comment (zhiltsov-max) [reply cvat/apps/webhooks/serializers.py]: Removed.

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py (hunk 1 above)
Comment (zhiltsov-max) [reply cvat/apps/dataset_manager/tests/test_rest_api_formats.py:2]: I've updated the headers

File: tests/python/rest_api/test_jobs.py [hunk 4]
```diff
@@ -146,6 +147,35 @@ def test_non_admin_list_jobs(
...
+class TestJobsListFilters(CollectionSimpleFilterTestBase):
+    field_lookups = {
+        "assignee": ["assignee", "username"],
+    }
+
+    @pytest.fixture(autouse=True)
+    def setup(self, restore_db_per_class, admin_user, jobs, tasks, projects):
+        self.user = admin_user
+        self.samples = jobs
+        self.sample_tasks = tasks
```
Comment (nmanovic) [inline tests/python/rest_api/test_jobs.py:159]: Do we need `sample_tasks` and `sample_projects` here?

File: cvat/apps/engine/views.py [hunk 5]
```diff
@@ -309,19 +308,21 @@ def perform_create(self, serializer, **kwargs):
...
+        deprecated=True) # TODO: remove in v2.5
     @action(detail=True, methods=['GET'], serializer_class=TaskReadSerializer,
         pagination_class=viewsets.GenericViewSet.pagination_class,
+        # These non-root list endpoints do not suppose extra options, just the basic output
         # Remove regular list() parameters from the swagger schema.
         # Unset, they would be taken from the enclosing class, which is wrong.
         # https://drf-spectacular.readthedocs.io/en/latest/faq.html#my-action-is-erroneously-paginated-or-has-filter-parameters-that-i-do-not-want
-        filter_fields=None, search_fields=None, ordering_fields=None)
+        filter_fields=None, ordering_fields=None, search_fields=None, simple_filters=None)
     def tasks(self, request, pk):
```
Comment (nmanovic) [inline cvat/apps/engine/views.py:321]: Let's remove all deprecated endpoints. They were changed (pagination). I see no reasons to keep them. In the future we will add {"projects": "link to list of tasks" }.

File: cvat/apps/engine/view_utils.py [hunk 6]
```diff
@@ -45,3 +51,46 @@ def make_paginated_response(
     serializer = serializer_type(queryset, **serializer_params)
 
     return response_type(serializer.data)
+
+def reverse(viewname, *, args=None, kwargs=None,
```
Comment (nmanovic) [inline cvat/apps/engine/view_utils.py:55]: The code below will be unnecessary if we delete old endpoints.

File: cvat/apps/engine/views.py (hunk 5 above)
Comment (zhiltsov-max) [reply cvat/apps/engine/views.py]: Done

**Pull request merged**